pkg/kds/v2/reconcile/snapshot_generator.go

```diff
@@ -40,9 +40,8 @@ type snapshotGenerator struct {
 	resourceMapper  reconcile.ResourceMapper
 }
 
-func (s *snapshotGenerator) GenerateSnapshot(ctx context.Context, node *envoy_core.Node) (envoy_cache.ResourceSnapshot, error) {
-	builder := cache_kds_v2.NewSnapshotBuilder()
-	for _, typ := range s.resourceTypes {
+func (s *snapshotGenerator) GenerateSnapshot(ctx context.Context, node *envoy_core.Node, builder cache_kds_v2.SnapshotBuilder, resTypes map[model.ResourceType]struct{}) (envoy_cache.ResourceSnapshot, error) {
```

Comment:
Nit: Could we split the params into separate lines? It starts to be hard to read so long line, especially during the code review
![image](https://github.com/kumahq/kuma/assets/11655498/fde7aa60-10aa-4415-824a-8e889df5e292)